### PR TITLE
Nsq to nsq

### PR DESCRIPTION
**Pull request by achille-roussel**
@thehydroimpulse 
@f2prateek 

Can you guys take a look at this?

### CODE REVIEW

File: cmd/nsq-to-nsq/main.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		LookupdHttpAddr        []string `conf:"lookupd-http-addr"         help:"List of nsqlookupd servers"`
+		DestinationNsqdTcpAddr []string `conf:"desintation-nsqd-tcp-addr" help:"List of nsqd nodes to publish to"`
+		NsqdTcpAddr            string   `conf:"nsqd-tcp-addr"             help:"Addres of the nsqd nodes to consume from"`
+		Topic                  string   `conf:"topic"                     help:"Topic to consume messages from"`
+		Channel                string   `conf:"channel"                   help:"Channel to consume messages from"`
+		DestinationTopic       string   `conf:"destination-topic"         help:"Topic to publish to"`
+		RateLimit              int      `conf:"rate-limit"                help:"Maximum number of message per second processed"`
+		MaxInFlight            int      `conf:"max-in-flight"             help:"Maximum number of in-flight messages"`
+	}{
+		LookupdHttpAddr:        []string{"localhost:4161"},
```
Comment (f2prateek) [inline cmd/nsq-to-nsq/main.go]: Should these be args?

**Comment by thehydroimpulse** [comment]
😍 LGTM!

File: cmd/nsq-to-nsq/main.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+func rateLimit(limit int, messages <-chan nsq.Message) <-chan nsq.Message {
+	if limit <= 0 {
+		return messages
+	}
+
+	output := make(chan nsq.Message, cap(messages))
+
+	go func() {
+		ticker := time.NewTicker(1 * time.Second)
```
Comment (f2prateek) [inline cmd/nsq-to-nsq/main.go]: Shouldn't this use the rate limit argument?

Comment (achille-roussel) [inline cmd/nsq-to-nsq/main.go]: No, the rate limit is in msg/sec, so the ticker fires every seconds to reset the count to 0, then the producer will simply stops if it reaches the limit within one second.

File: cmd/nsq-to-nsq/main.go (hunk 1 above)
Comment (achille-roussel) [inline cmd/nsq-to-nsq/main.go]: What do you mean?

Comment (f2prateek) [inline cmd/nsq-to-nsq/main.go]: As in they should be read from the config?

Comment (f2prateek) [inline cmd/nsq-to-nsq/main.go]: Oh I see, these are just the defaults.

**Comment by f2prateek** [comment]
LGTM

File: cmd/nsq-to-nsq/main.go (hunk 1 above)
Comment (achille-roussel) [inline cmd/nsq-to-nsq/main.go]: Yep ;)

**Pull request merged**